### PR TITLE
Updates Lemat Revolver, adds 20ga percussion, and abstracts cap and ball revolvers

### PR DESCRIPTION
**Pull request by tenmillimaster**
#### Summary
```SUMMARY: Content "Adds new 20ga percussion and updates C&B revolvers'"```

#### Purpose of change
The LeMat revolver currently uses.44 magnum and 20 ga shells. Historically this was a .42/.35 cal muzzleloader, and only briefly got converted to pinfire and centerfire metallic cartridges before production ended.

Reload times on the revolver felt a little fast.
#### Describe the solution

I timed someone on youtube performing **just** the capping, insertion and ramming actions, so made the reloads take about 1.5 seconds. I've added an abstract cap and ball revolver, and updated the other revolvers to draw from this abstract. The abstract swaps the sights rail for a sight mount slot, and precludes installation of underbarrel mods- that's where the loading lever goes.

~~The `STR_RELOAD` flag has been added to this abstract revolver, as one would be running the ramrod under stress potentially. At 8 strength, 0 skills, it  should be a 160 sec speed reduction (so don't try to reload cap and ball if you're at 2 strength or something).~~

I ended up just ditching str reload and setting the reload time to 1.5 seconds.

Using a technique I have been working on, I've updated the volume of the LeMat based on images and known references with ImageJ.

The Lemat is now the Pietta reproduction chambered in .44 cal and 20ga. 20 gauge paper cartridges have been added as an ammotype, with the damage set specifically for normal length percussion shotguns, hence the lemat shotgun's large damage penalty.  I was unable to find empirically determined damages for 20 gauge barrel lengths, so have approximated the damage based on [empirically determined 12 gauge barrel lengths](https://www.smallarmsreview.com/display.article.cfm?idarticles=1410).  Data [here](https://www.smallarmsreview.com/images/articles/art_000960/005.jpg).  This reduction was on top of the one that black powder 12 ga shotgun shells had.

#### Describe alternatives you've considered
Not being me.

#### Additional context
<https://imagej.nih.gov/ij/docs/guide/146-30.html>

The feret angle and diameter are defined, above, as follows:
```
Feret’s diameter The longest distance between any two points along the selection boundary, also known as maximum caliper. Uses the heading Feret. The angle (0--180 degrees) of the Feret’s diameter is displayed as FeretAngle [...]
```

 
Line 1. Uncalibrated analysis of base image, post Feret angle rotation. You will notice that this feret angle is approximately 180 degrees. The image is rotated to it's longest drawable line (the Feret distance), which is also our firearm's length.

 
The image is then calibrated with 356.004 pixels = [340.36mm](https://www.taylorsfirearms.com/hand-guns/blackpowder-revolvers/le-mat-cavalry/le-mat-cavalry.html)

Line  2. Re analyized base image- the area displayed is now in square mm.

Line 3. I was unable to find the width of the lemat, but did find a rather good image of its cylinder from the same manufacturer. 
Post processing image:
 

The representative red line represents the diameter of the chamber in pixels, which in line 3 is feret distance 112.894 pixels. the image is calibrated with this = known distance of the diameter of a .44cal pietta chamber.

Line 4. the whole cylinder was analyzed, this time yielding feret distance of 45.527 mm (due to above calibration). Represented as the 'straight' blue line. Thus we'll use a 46mm width for the lemat's volume calc.

Volume is = area of revolver x width of cylinder.

### CODE REVIEW

**Comment by tenmillimaster** [comment]
Having trouble finding good sources on what 20ga/.62 cal will do out of a <6" barrel, but the closest I've found is here: https://www.muzzleloadingforum.com/threads/estimated-mv.62491/?

325gr(.600 lead ball) at 910 fps for 808 joules, or 28.4 damage.

**Comment by tenmillimaster** [comment]
The reload times aren't matching what I thought they'd be, I need to poke around this some more.

**Comment by tenmillimaster** [comment]
`STR_RELOAD` ditched since I only wanted it to be a minor buff/debuff, it doesn't seem to be doing anything, and the scaling is too high anyways.

File: data/json/items/gunmod/underbarrel.json [hunk 1]
```diff
@@ -157,25 +157,25 @@
...
       "range": -2,
-      "ranged_damage": 2,
+      "ranged_damage": -12,
       "dispersion": 600,
       "durability": 10,
+      "reload": 310,
       "clip_size": 1
     },
-    "flags": [ "RELOAD_ONE", "IRREMOVABLE" ]
+    "flags": [ "RELOAD_ONE", "NO_UNLOAD", "STR_RELOAD", "IRREMOVABLE" ]
```
Comment (Rail-Runner) [inline data/json/items/gunmod/underbarrel.json]: STR_RELOAD is still present here; not sure if intended.

**Pull request merged**